### DISCUSSION TITLE
Styling

### DISCUSSION THREAD

**Discussion by djelenc** [rharish101/ReGreet#74] [General]
I'd like to style the log-in screen with CSS, but I do not know which selectors to use. For instance, how do I style the background color? What are the other (major) elements that I can select and change from within CSS?

**Comment by max-ishere** [comment]
You can get the selectors from the GTK inspector. Press <kbd>Ctrl</kbd> <kbd>Shift</kbd> <kbd>D</kbd> to open it. You can either use demo mode (`regreet --demo --style ./style.css`) or fakegreet (`fakegreet 'regreet --style ./style.css'`) to run ReGreet as your user.

Then using the inspector select the element you want (there's a button for it) and click on CSS properties. Here you can see the path for an element.

Here's a screenshot from [the GTK docs](https://developer.gnome.org/documentation/tools/inspector.html).

![](https://developer.gnome.org/documentation/_images/inspector-objects-css-nodes-light.png)

The selector goes as follows: `name#id.style-classes`

Here are GTK4 CSS docs:
https://docs.gtk.org/gtk4/css-overview.html

And you can also use any pre made theme if you want, but I think you're going for a custom thing so that may not work out.

Hope this was helpful. There was a PR to make styling easier #73, but looks like it got canceled.

**Reply by djelenc** [reply]
Thank you so much!

**Reply by max-ishere** [reply]
You're welcome. You can mark it as a solution then.
